pydantic/fields.py

```diff
@@ -1083,7 +1083,7 @@ def _validate_discriminated_union(
         except TypeError:
             try:
                 # BaseModel or dataclass
-                discriminator_value = getattr(v, self.discriminator_alias)
+                discriminator_value = getattr(v, self.discriminator_key)
```

Comment:
What if someone writes `discriminator='lit'`? I remember this comment https://github.com/samuelcolvin/pydantic/pull/2336#issuecomment-913498533, which I started to tackle by checking that the alias was the same. But in fact it probably makes sense to only check the field key, in which case you're absolutely right.